<launch>

    <node pkg="coax_server"
          name="coax"
          type="coax_server"
          output="screen"
          args="/dev/ttyS0:1">
      <remap from="/coax_server/state" to="/state"/>

      <remap from="/coax_server/control" to="/control"/>
      <remap from="/coax_server/rawcontrol" to="/rawcontrol"/>

      <remap from="/coax_server/reach_nav_state" to="/reach_nav_state"/>
      <remap from="/coax_server/configure_comm" to="/configure_comm"/>
      <remap from="/coax_server/set_timeout" to="/set_timeout"/>
    </node>

    <node pkg="coax_simple_control"
          name="CoaxSimpleControl"
          type="CoaxSimpleControl"
          output="screen">
      <remap from="/CoaxSimpleControl/state" to="/state"/>
      <remap from="/CoaxSimpleControl/tf" to="/tf"/>      

      <remap from="/CoaxSimpleControl/rawcontrol" to="/rawcontrol"/>
      <remap from="/CoaxSimpleControl/simplecontrol" to="/control"/>
      <remap from="/CoaxSimpleControl/vel_approx" to="/vel_approx"/>

      <remap from="/CoaxSimpleControl/reach_nav_state" to="/reach_nav_state"/>
      <remap from="/CoaxSimpleControl/configure_comm" to="/configure_comm"/>
      <remap from="/CoaxSimpleControl/configure_control" to="/configure_control"/>
      <remap from="/CoaxSimpleControl/set_timeout" to="/set_timeout"/>

      <remap from="/CoaxSimpleControl/set_nav_mode" to="/set_nav_mode"/>
      <remap from="/CoaxSimpleControl/set_control_mode" to="/set_control_mode"/>
      <remap from="/CoaxSimpleControl/set_waypoint" to="/set_waypoint"/>

      <param name="frequency" value="100"/>

      <rosparam file="$(find coax_simple_control)/config/coax_parameters.yaml"/>
      <rosparam file="$(find coax_simple_control)/config/coax_control_params.yaml"/>

    </node>

</launch>
